In [50]:
import numpy as np
import pandas as pd

**1. Peaks**

a) What proportion of peaks is still unclimbed?

In [51]:
peaks = pd.read_csv("data/peaks.csv")

In [52]:
peaks.shape

(468, 8)

In [53]:
peaks.head()

peak_id      peak_name peak_alternative_name  height_metres climbing_status  \
0    AMAD     Ama Dablam          Amai Dablang           6814         Climbed   
1    AMPG  Amphu Gyabjen                   NaN           5630         Climbed   
2    ANN1    Annapurna I                   NaN           8091         Climbed   
3    ANN2   Annapurna II                   NaN           7937         Climbed   
4    ANN3  Annapurna III                   NaN           7555         Climbed   

   first_ascent_year  first_ascent_country first_ascent_expedition_id  
0             1961.0  New Zealand, USA, UK                  AMAD61101  
1             1953.0                    UK                  AMPG53101  
2             1950.0                France                  ANN150101  
3             1960.0             UK, Nepal                  ANN260101  
4             1961.0                 India                  ANN361101

In [54]:
peaks_unclimbed = peaks[peaks['climbing_status'] == 'Unclimbed']
peaks_unclimbed.describe()

height_metres  first_ascent_year
count     127.000000                1.0
mean     6523.330709             2018.0
std       407.684422                NaN
min      5407.000000             2018.0
25%      6251.000000             2018.0
50%      6500.000000             2018.0
75%      6690.500000             2018.0
max      8077.000000             2018.0

Answer: <br>
Unclimed Peaks/All Peaks = 127/468 = 0.2713 = 27.13%<br>
The proportion of peaks unclimbed is 27.13%.<br>

b) What is the average height of the climbed peaks vs. unclimbed peaks?

In [55]:
peaks_climbed = peaks[peaks['climbing_status'] == 'Climbed']
peaks_climbed.describe()

height_metres  first_ascent_year
count     341.000000         335.000000
mean     6706.284457        1978.961194
std       615.268778         100.332688
min      5550.000000         201.000000
25%      6227.000000        1963.000000
50%      6609.000000        1982.000000
75%      7030.000000        2008.000000
max      8850.000000        2019.000000

Answer: <br>
The result shows no outliers outliers in the data of the height of the peaks.Therefore, we can draw some conclusions from it.<br>
The average height of the climbed peaks is 6706.28 meters and the average height of the unclimbed peaks is 6523.33 meters.

**2. Sherpas**

In [56]:
members = pd.read_csv("data/members.csv")

In [57]:
members.head()

expedition_id     member_id peak_id   peak_name  year  season sex   age  \
0     AMAD78301  AMAD78301-01    AMAD  Ama Dablam  1978  Autumn   M  40.0   
1     AMAD78301  AMAD78301-02    AMAD  Ama Dablam  1978  Autumn   M  41.0   
2     AMAD78301  AMAD78301-03    AMAD  Ama Dablam  1978  Autumn   M  27.0   
3     AMAD78301  AMAD78301-04    AMAD  Ama Dablam  1978  Autumn   M  40.0   
4     AMAD78301  AMAD78301-05    AMAD  Ama Dablam  1978  Autumn   M  34.0   

  citizenship expedition_role  ...  highpoint_metres  success   solo  \
0      France          Leader  ...               NaN    False  False   
1      France   Deputy Leader  ...            6000.0    False  False   
2      France         Climber  ...               NaN    False  False   
3      France      Exp Doctor  ...            6000.0    False  False   
4      France         Climber  ...               NaN    False  False   

   oxygen_used   died  death_cause death_height_metres  injured  injury_type  \
0        False  False          NaN                 NaN    False          NaN   
1        False  False          NaN                 NaN    False          NaN   
2        False  False          NaN                 NaN    False          NaN   
3        False  False          NaN                 NaN    False          NaN   
4        False  False          NaN                 NaN    False          NaN   

  injury_height_metres  
0                  NaN  
1                  NaN  
2                  NaN  
3                  NaN  
4                  NaN  

[5 rows x 21 columns]

Using the members data only, select a subset of all persons who have a Nepalese citizenship.

In [58]:
Nepalese = members[members['citizenship'] == "Nepal"]

a) What proportion of them was hired for the expedition?

In [59]:
Nepalese.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 16135 entries, 24 to 76518
Data columns (total 21 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   expedition_id         16135 non-null  object 
 1   member_id             16135 non-null  object 
 2   peak_id               16135 non-null  object 
 3   peak_name             16131 non-null  object 
 4   year                  16135 non-null  int64  
 5   season                16135 non-null  object 
 6   sex                   16135 non-null  object 
 7   age                   14373 non-null  float64
 8   citizenship           16135 non-null  object 
 9   expedition_role       16133 non-null  object 
 10  hired                 16135 non-null  bool   
 11  highpoint_metres      14247 non-null  float64
 12  success               16135 non-null  bool   
 13  solo                  16135 non-null  bool   
 14  oxygen_used           16135 non-null  bool   
 15  died              

In [60]:
NepaleseHired = Nepalese[Nepalese['expedition_role'] == 'H-A Worker']
NepaleseHired.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 14279 entries, 24 to 76518
Data columns (total 21 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   expedition_id         14279 non-null  object 
 1   member_id             14279 non-null  object 
 2   peak_id               14279 non-null  object 
 3   peak_name             14277 non-null  object 
 4   year                  14279 non-null  int64  
 5   season                14279 non-null  object 
 6   sex                   14279 non-null  object 
 7   age                   12877 non-null  float64
 8   citizenship           14279 non-null  object 
 9   expedition_role       14279 non-null  object 
 10  hired                 14279 non-null  bool   
 11  highpoint_metres      13148 non-null  float64
 12  success               14279 non-null  bool   
 13  solo                  14279 non-null  bool   
 14  oxygen_used           14279 non-null  bool   
 15  died              

Answer:<br>
Hired Nepalese / All Nepalese = 14279/16135 = 0.8850 = 88.5%<br>
The proportion of Nepalese members who were hired for expedition is 88.5%.

b) What are their minimum, maximum, and average ages?

In [61]:
Nepalese.describe()

year           age  highpoint_metres  death_height_metres  \
count  16135.000000  14373.000000      14247.000000           297.000000   
mean    2005.505671     32.885062       7878.236190          6059.461279   
std       14.479732      7.746973        987.588329          1277.859557   
min     1921.000000     12.000000       3800.000000          2100.000000   
25%     2002.000000     27.000000       6977.000000          5350.000000   
50%     2010.000000     32.000000       8188.000000          6000.000000   
75%     2015.000000     38.000000       8850.000000          6800.000000   
max     2019.000000     85.000000       8850.000000          8800.000000   

       injury_height_metres  
count            122.000000  
mean            6886.147541  
std             1116.734982  
min             4600.000000  
25%             6050.000000  
50%             6900.000000  
75%             7800.000000  
max             8880.000000

In [62]:
NepaleseHired.describe()

year           age  highpoint_metres  death_height_metres  \
count  14279.000000  12877.000000      13148.000000           213.000000   
mean    2006.776035     32.797701       7902.146562          6449.483568   
std       13.480210      7.679750        974.865263          1036.122906   
min     1921.000000     15.000000       3800.000000          4200.000000   
25%     2004.000000     27.000000       7000.000000          5790.000000   
50%     2010.000000     32.000000       8188.000000          6400.000000   
75%     2015.000000     38.000000       8850.000000          7000.000000   
max     2019.000000     67.000000       8850.000000          8800.000000   

       injury_height_metres  
count             97.000000  
mean            6845.515464  
std             1081.711964  
min             4600.000000  
25%             6050.000000  
50%             6950.000000  
75%             7500.000000  
max             8880.000000

Answer:<br>
The minimum age of all Nepalese members is 12 years old. The maximum age of all Nepalese members is 85 years old. The average age of all Nepalese members is about 33 years old.<br> 
The minimum age of hired Nepalese members is 15. The maximum age of hired Nepalese members is 67. The average age of hired Nepalese members is about 33 years old.

**3. Gender**

Now use all three data sets and join them on the provided ID variables. For this question, limit yourself to the subset of non-hired participants.

In [63]:
expeditions = pd.read_csv("data/expeditions.csv")

In [64]:
t1 = pd.merge(expeditions,peaks,how = 'left',on = 'peak_id')

In [65]:
non_hired = members[members['hired'] == False]

In [66]:
t2 = pd.merge(non_hired,t1,how = "left", on = "expedition_id")

In [73]:
t2.loc[:,['sex','success']].groupby('sex').mean()

success
sex          
F    0.326583
M    0.312743

a) What is the first record of a woman summitting a peak? Which peak (name and height in meters) was that?

In [67]:
selected_members = members[['expedition_id','peak_name','year','sex','expedition_role','highpoint_metres','success']].drop_duplicates()
female_members = selected_members[selected_members['sex'] == "F" ]
female_members_success = female_members[female_members['success'] == True ]
female_members_success.head(2)

expedition_id   peak_name  year sex expedition_role  highpoint_metres  \
108     AMAD82101  Ama Dablam  1982   F         Climber            6814.0   
110     AMAD82101  Ama Dablam  1982   F          Leader            6814.0   

     success  
108     True  
110     True

In [68]:
female_members_success.sort_values(by = ['year'])

expedition_id         peak_name  year sex expedition_role  \
31468     DZAN49101            Dzanye  1949   F         Climber   
24739     TENG49101          Tengkoma  1949   F         Climber   
22264     GAN155301          Ganesh I  1955   F         Climber   
22230     LEOE55101  Leonpo Gang East  1955   F          Leader   
22232     LEOE55101  Leonpo Gang East  1955   F         Climber   
...             ...               ...   ...  ..             ...   
75522     EVER19126           Everest  2019   F         Climber   
75521     EVER19126           Everest  2019   F          Leader   
75516     EVER19125           Everest  2019   F         Climber   
75465     EVER19122           Everest  2019   F         Climber   
76467     MAKA19107            Makalu  2019   F          Leader   

       highpoint_metres  success  
31468            6719.0     True  
24739            6215.0     True  
22264            7422.0     True  
22230            6733.0     True  
22232            6733.0     True  
...                 ...      ...  
75522            8850.0     True  
75521            8850.0     True  
75516            8850.0     True  
75465            8850.0     True  
76467            8485.0     True  

[1643 rows x 7 columns]

Answer:<bk>
The first record of a woman summitting a peak is in the year 1949. 

b) Provide a single cross-tab table of the sex of an expedition participant and - the proportion of success in summitting a main peak or sub-peak
- the proportion of participants using oxygen support - the proportion of participants who died.

That is, sex should be displayed as the rows of the output and the three requested proportions as its columns.

In [75]:
t2.loc[:,['sex','success','oxygen_used']].groupby('sex').mean()

KeyError: "['oxygen_used'] not in index"

In [70]:
male_members = members[members['sex'] == "M" ]
male_members_success = male_members[male_members['success'] == True ]
male_members.info()
male_members_success.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 69473 entries, 0 to 76518
Data columns (total 21 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   expedition_id         69473 non-null  object 
 1   member_id             69473 non-null  object 
 2   peak_id               69473 non-null  object 
 3   peak_name             69458 non-null  object 
 4   year                  69473 non-null  int64  
 5   season                69473 non-null  object 
 6   sex                   69473 non-null  object 
 7   age                   66186 non-null  float64
 8   citizenship           69463 non-null  object 
 9   expedition_role       69454 non-null  object 
 10  hired                 69473 non-null  bool   
 11  highpoint_metres      49984 non-null  float64
 12  success               69473 non-null  bool   
 13  solo                  69473 non-null  bool   
 14  oxygen_used           69473 non-null  bool   
 15  died               